python/lsst/analysis/drp/histPlot.py

```diff
@@ -274,94 +236,95 @@ def histPlot(self, catPlot, plotInfo, sumStats):
-            rightPanelAx.axvline(datMed, ls=':', lw=2, c=col)
-        rightPanelAx.set_xlim(rightPanelLower, rightPanelUpper)
-        rightPanelAx.set_xlabel(self.config.axisLabels["xRight"])
-        rightPanelAx.tick_params(labelsize=7)
-        yStep = 0.038*len(self.config.leftPanelActions.fieldNames) + 0.01
-        rightPanelAx.legend(loc="upper left", bbox_to_anchor=(0.00, 0.77-yStep),
-                            bbox_transform=fig.transFigure, fontsize=7, handleheight=1.5)
-        rightPanelAx.axvline(0, ls='--', lw=2, c='k')
-
-        # Corner plot of patches showing summary stat in each
-        axCorner = plt.gcf().add_subplot(gs[70:96, 3:21])
+        gs = gridspec.GridSpec(240, 240)
+
+        # determine gridspec divisions
+        summary_block = 40
+        panel_pad = 8
+        if len(panels) <= 2:
+            ncols = 1
+        else:
+            ncols = 2
+        nrows = int(np.ceil(len(panels) / ncols))
+        col_bounds = np.linspace(0+summary_block+panel_pad, 240, ncols+1, dtype=int)
+        row_bounds = np.linspace(0, 240, nrows+1, dtype=int)
+        col_starts, col_stops, row_starts, row_stops = [], [], [], []
+        for i in range(len(panels)):
+            col_starts.append(col_bounds[i%ncols]+panel_pad)
+            if (i == (len(panels)-1)) and (len(panels)%2 == 1):
+                col_stops.append(np.max(col_bounds)-panel_pad)
+            else:
+                col_stops.append(col_bounds[(i%ncols)+1]-panel_pad)
+            row_starts.append(row_bounds[i//ncols]+panel_pad)
+            row_stops.append(row_bounds[(i//ncols)+1]-(2*panel_pad))
+
+        # panel plotting
+        for i, (panel, col_start, col_stop, row_start, row_stop) in enumerate(zip(panels,
+                                                                                  col_starts, col_stops,
+                                                                                  row_starts, row_stops)):
+            hists = [x for x in panels[panel].toDict()['actions'].keys()]
+            columns = []
+            for hist in hists:
+                columns.append(f"p{i}_{hist}")
+
+            # panel plotting limits
+            vLowers, vUppers = [], []
+            for column in columns:
+                pvals = np.nanpercentile(datPlot[column].values,
+                                         [self.config.panels[panel].pLower, self.config.panels[panel].pUpper])
+                vLowers.append(pvals[0])
+                vUppers.append(pvals[1])
+            vLower = np.min(vLowers)
+            vUpper = np.max(vUppers)
+
+            # generate plot
+            ax = fig.add_subplot(gs[row_start:row_stop, col_start:col_stop])
+            for count, (hist, column) in enumerate(zip(hists, columns)):
+                isfinite = np.isfinite(datPlot[column])
+                datMed = np.nanmedian(datPlot[column][isfinite])
+                datMad = sigmaMad(datPlot[column][isfinite])
```

Comment:
You should omit the NaNs from this.

Reply:
The median values are now calculated further up the script, alongside the x-axis limits. The relevant median line now looks like this:
```
meds.append(np.nanmedian(hist_data))
```
I've removed the `[isfinite]` selection from the data selector, so use of `nanmedian` is required in case any non-finite values creep into the calculation.

Reply:
Nevermind, I went back to using the `isfinite` logic. This section now looks like this:
```
isfinite = np.isfinite(hist_data)
meds.append(np.median(hist_data[isfinite]))
mads.append(sigmaMad(hist_data[isfinite]))
nums.append(np.sum(isfinite))
```